File: IslandPerimeter/IslandPerimeter.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var islandPerimeter = function (grid) {
  let rows = grid.length;
  let cols = grid[0].length;
  let perimeter = 0;
  for (let row = 0; row < rows; row++) {
    for (let col = 0; col < cols; col++) {
      if (!grid[row][col]) continue;
      perimeter += 4;
      if (row > 0 && grid[row - 1][col]) perimeter--;
      if (col > 0 && grid[row][col - 1]) perimeter--;
      if (row < rows - 1 && grid[row + 1][col]) perimeter--;
      if (col < cols - 1 && grid[row][col + 1]) perimeter--;
    }
  }
  return perimeter;
};

File: IslandPerimeter/IslandPerimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        land, nei = 0, 0
        for i in range(m):
            for j in range(n):
                if grid[i][j] == 1:
                    land += 1
                    if i < m-1 and grid[i+1][j]==1:
                        nei+=1
                    if j < n-1 and grid[i][j+1]==1:
                        nei+=1
        return land * 4 - 2 * nei
